src/backend/vulkan/src/lib.rs

```diff
@@ -959,8 +960,12 @@ impl hal::queue::RawCommandQueue<Backend> for CommandQueue {
         };
 
         match self.swapchain_fn.queue_present_khr(*self.raw, &info) {
-            vk::Result::SUCCESS => Ok(()),
-            vk::Result::SUBOPTIMAL_KHR | vk::Result::ERROR_OUT_OF_DATE_KHR => Err(()),
+            vk::Result::SUBOPTIMAL_KHR | vk::Result::SUCCESS => Ok(()),
```

Comment:
Should we return Suboptimal as part of the Ok(()) in case user wants to handle it?